spec/grocer_spec.rb

```diff
@@ -22,296 +23,311 @@
+      cart = [find_item_by_name_in_collection('TEMPEH', items)]
       consolidated_cart = consolidate_cart(cart)
 
-      result = apply_clearance(consolidated_cart)
-      expect(result["TEMPEH"][:price]).to eq(2.40)
+      apply_clearance(consolidated_cart)
+      expect(consolidated_cart.first[:price]).to be_within(0.1).of(2.40)
     end
 
     it "does not discount the price for items not on clearance" do
-      cart = [find_item('AVOCADO'), find_item('TEMPEH'), find_item('BEETS'), find_item('SOY MILK')]
+      cart = [
+        find_item_by_name_in_collection('AVOCADO', items),
+        find_item_by_name_in_collection('TEMPEH', items),
+        find_item_by_name_in_collection('BEETS', items),
+        find_item_by_name_in_collection('SOY MILK', items)
+      ]
       consolidated_cart = consolidate_cart(cart)
-      result = apply_clearance(consolidated_cart)
+      apply_clearance(consolidated_cart)
       clearance_prices = {"AVOCADO" => 2.40, "TEMPEH" => 2.40, "BEETS" => 2.50, "SOY MILK" => 3.60}
-      result.each do |name, properties|
-        expect(properties[:price]).to eq(clearance_prices[name])
+
+      i = 0
+      while  i < consolidated_cart.length do
+        item = consolidated_cart[i]
+        expect(item[:price]).to be_within(0.1).of(clearance_prices[item[:item]])
+        i += 1
       end
     end
   end
 
   describe "#checkout" do
-
-    describe "base case (no clearance, no coupons)" do
+    describe "in base case (no clearance, no coupons)" do
       it "calls on #consolidate_cart before calculating the total for one item" do
-        cart = [find_item('BEETS')]
-        result = consolidate_cart(cart)
-
-        expect(self).to receive(:consolidate_cart).with(cart).and_return(result)
+        cart = [find_item_by_name_in_collection('BEETS', items)]
         expect(checkout(cart, [])).to eq(2.50)
       end
 
       it "calls on #apply_coupons after calling on #consolidate_cart when there is only one item in the cart" do
-        cart = [find_item('BEETS')]
+        cart = [find_item_by_name_in_collection('BEETS', items)]
 
         consolidated = consolidate_cart(cart)
-        coupons_applied = apply_coupons(consolidated, [])
-
-        expect(self).to receive(:consolidate_cart).with(cart).and_return(consolidated)
-        expect(self).to receive(:apply_coupons).with(consolidated, []).and_return(coupons_applied)
+        apply_coupons(consolidated, [])
 
         expect(checkout(cart, [])).to eq(2.50)
       end
 
       it "calls on #apply_clearance after calling on #apply_coupons when there is only one item in the cart and no coupon" do
-        cart = [find_item('BEETS')]
+        cart = [find_item_by_name_in_collection('BEETS', items)]
 
         consolidated = consolidate_cart(cart)
-        coupons_applied = apply_coupons(consolidated, [])
-        clearance_applied = apply_clearance(coupons_applied)
+        apply_coupons(consolidated, [])
+        apply_clearance(consolidated)
```

Comment:
these lines and 257-259 were causing some problems while testing this - I've changed them to match the expected (previously, these methods returned values that were then passed on to the next, but now the lab requires students to modify the existing array, which would cause the other `apply_clearance` tests to pass but fail when using the method here or vice versa.

It looks though like these aren't being used, or am I wrong? The expectation is just running `checkout`, and not using whatever these lines are calculating to compare

Reply:
I don't think that's right. In both of these items, the method is updating what's passed in.

https://github.com/learn-co-curriculum/ruby-enumerables-hash-practice-green-grocer-lab/blob/solution/grocer.rb#L17-L36

https://github.com/learn-co-curriculum/ruby-enumerables-hash-practice-green-grocer-lab/blob/solution/grocer.rb#L38-L45

Reply:
the instructions in the new starter code are to modify the existing array and the 2.01 solution for `checkout` modifies the cart without returning anything: https://github.com/learn-co-curriculum/programming-univbasics-nds-green-grocer/blob/wip-v2.01-solution/grocer.rb

It is possible with this set up to pass the `apply_clearance` tests without having to return anything, but then fail the tests here because they're expecting a return

Reply:
so this aligns with the new solution

Reply:
Also other parts of the tests: https://github.com/learn-co-curriculum/programming-univbasics-nds-green-grocer/blob/wip-v2.01/spec/grocer_spec.rb#L76-L87